eng/tools/summarize-impact/src/impact.ts

```diff
@@ -0,0 +1,917 @@
+    breakingChangesCheckType.CrossVersion.reviewRequiredLabel,
+    labelContext.present,
+  );
+
+  const versioningApprovalLabels = breakingChangesCheckType.SameVersion.approvalLabels.map(
+    (label: string) => new Label(label, labelContext.present),
+  );
+  const breakingChangeApprovalLabels = breakingChangesCheckType.CrossVersion.approvalLabels.map(
+    (label: string) => new Label(label, labelContext.present),
+  );
+
+  // ----- Breaking changes label processing logic -----
+  // These lines implement the set of rules determining which of the breaking change
+  // labels should be present on given PR.
+  // The doc describing breaking change review rules can be found here:
+  // https://aka.ms/brch-dev
+
+  // ❗ IMPORTANT ❗: this MUST be set BEFORE versioningReviewRequiredLabel.shouldBePresent
+  // due to logical dependency
+  breakingChangeReviewRequiredLabel.shouldBePresent =
+    // "BreakingChangeReviewRequired" label should be present if
+    // 1. The PR targets a production branch
+    prTargetsProductionBranch &&
+    // 2. AND given OAD run determined it is still applicable.
+    breakingChangesLabelsFromOad.includes(breakingChangeReviewRequiredLabel.name);
+
+  // ❗ IMPORTANT ❗: this MUST be set AFTER breakingChangeReviewRequiredLabel.shouldBePresent
+  // due to logical dependency
+  versioningReviewRequiredLabel.shouldBePresent =
+    // "VersioningReviewRequired" label should be unconditionally removed if
+    // the label "BreakingChangeReviewRequired" should be present.
+    !breakingChangeReviewRequiredLabel.shouldBePresent &&
+    // AND "VersioningReviewRequired" label should be present if
+    // 1. The PR targets a production branch
+    prTargetsProductionBranch &&
+    // 2. AND given OAD run determined it is still applicable.
+    breakingChangesLabelsFromOad.includes(versioningReviewRequiredLabel.name);
+
+  versioningApprovalLabels.forEach((approvalLabel) => {
+    approvalLabel.shouldBePresent =
+      approvalLabel.present && versioningReviewRequiredLabel.shouldBePresent;
+  });
+
+  breakingChangeApprovalLabels.forEach((approvalLabel) => {
+    approvalLabel.shouldBePresent =
+      approvalLabel.present && breakingChangeReviewRequiredLabel.shouldBePresent;
+  });
+
+  // ----------
+
+  applyLabelsStateChanges(labelContext.toAdd, labelContext.toRemove);
+
+  logDiagInfo();
+
+  return {
+    versioningReviewRequiredLabelShouldBePresent: versioningReviewRequiredLabel.shouldBePresent,
+    breakingChangeReviewRequiredLabelShouldBePresent:
+      breakingChangeReviewRequiredLabel.shouldBePresent,
+  };
+
+  /**
+   * Get labels denoting required breaking change or versioning review.
+   * The labels are read from ADO pipeline environment variables.
+   * These variables are set to appropriate value by BreakingChangesRuleManager.addBreakingChangeLabels().
+   * Read that function's comment for details.
+   */
+  function getBreakingChangesLabelsFromOad(): string[] {
```

Comment:
Right here.

Reply:
This entire section will be eliminated, as post-breakingChange run we trigger `Update-Labels`.